socks5proxy/src/socks5client.cpp

```diff
@@ -0,0 +1,337 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "socks5client.h"
+#include "socks5.h"
+
+#ifdef Q_OS_WIN
+#  include <winsock2.h>
+#  include <ws2ipdef.h>
+#else
+#  include <arpa/inet.h>
+#endif
+
+#include <QHostAddress>
+
+namespace {
+
+#ifdef Q_OS_WIN
+#  define PACK(__Declaration__) \
+    __pragma(pack(push, 1)) __Declaration__ __pragma(pack(pop))
+#else
+#  define PACK(__Declaration__) __Declaration__ __attribute__((__packed__))
+#endif
+
+PACK(struct ClientGreetingPacket {
+  uint8_t version;
+  uint8_t nauth;
+});
+
+PACK(struct ServerChoicePacket {
+  uint8_t version;
+  uint8_t cauth;
+});
+
+PACK(struct ClientConnectionRequestPacket {
+  uint8_t version;
+  uint8_t cmd;
+  uint8_t rsv;
+  uint8_t atype;
+});
+
+PACK(struct ClientConnectionAddressIpv4Packet {
+  uint8_t ip_dst[4];
+  uint16_t port;
+});
+
+PACK(struct ClientConnectionAddressDomainNamePacket {
+  uint8_t cmd;
+  uint16_t port;
+});
+
+PACK(struct ClientConnectionAddressIpv6Packet {
+  uint8_t ip_dst[16];
+  uint16_t port;
+});
+
+PACK(struct ServerResponsePacket {
+  uint8_t m_version = 0x05;
+  uint8_t m_rep = 0x01;
```

Comment:
For cleanliness, I feel like we need to add an enumeration for the response codes instead of using hardcoded values and comments spread across this file.